### PR TITLE
fix: button variant table binding

### LINKED ISSUE #8529: [Bug]: Button variant binded with text widget does not populate on the table icon column
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

It is observed that the button variant options passed with text widget Text label as PRIMARY, SECONDARY and TERTIARY and binded with table icon type - Button variant - {{Text1.text}} throws error

[LOOM DEMO](https://www.loom.com/share/baa63d3204554faf8016d2bee6ca7b27)

### Steps To Reproduce

```markdown
1.Launch Appmsith app in release/Prod Environment 
2.Drag and drop text widget and pass the Value as PRIMARY, SECONDARY or TERTIARY 
3.Drag and Drop table widget 
4.Change the column type to column 
5.Click on Button variant and enable JS and pass the following syntax {{Text1.text}} and observe
```

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

Button variant binded with text widget does not populate on the table icon column 

Fixes #8529

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

File: app/client/src/workers/validations.ts [hunk 1]
```diff
@@ -862,4 +863,58 @@ export const VALIDATORS: Record<ValidationTypes, Validator> = {
       return invalidResponse;
     }
   },
+  [ValidationTypes.TEXT_OR_ARRAY]: (
```
Comment (riodeuno) [inline app/client/src/workers/validations.ts:866]: Please add unit tests for this validation.

**Review by riodeuno — CHANGES_REQUESTED** [review]

**Comment by riodeuno** [comment]
An ideal solution here would be to disallow multiple types of values for a property. If it is text or array , it can be only array with single or more values.

**Review by riodeuno — DISMISSED** [review]

**Comment by riodeuno** [comment]
@aswathkk Please review this change. Adding you as a reviewer. Looks good from my end.

**Comment by aswathkk** [comment]
@riodeuno I found some issues in the DP.
What I would suggest is to wait for #9396 to be merged so that we can use `ValidationTypes.TABLE_PROPERTY` to write a simple validation that is going to fix #8529

**Review by aswathkk — APPROVED** [review]
LGTM

**Comment by YogeshJayaseelan** [comment]
@Tooluloope Verified this fix and it is working as expected

**Comment by wealthz01** [comment]
Hello there ... pls can i know how to get started with this online chat or trading

**Review by riodeuno — APPROVED** [review]

**Pull request merged**